### DISCUSSION TITLE
2.0.0

### DISCUSSION THREAD

**Discussion by rtivital** [mantinedev/mantine#125] [Releases]
# Mantine 2.0.0

[View changelog with demos on Mantine website](https://mantine.dev/pages/changelog)

## Breaking changes

- All inputs no longer support `inputStyle` and `inputClassName` props, this feature was replaced by [Styles API](https://mantine.dev/theming/styles-api/)
- Burger component no longer uses colors from theme, use
  hex color value instead
- Select component was renamed to NativeSelect
- `ElementsGroup`, `Hr` and `CardsGrid` components are replaced and no longer exported from @mantine/core package
- All inputs variant is set by default to filled in dark theme

## New theming options

- @mantine/core package now exports [GlobalStyles and NormalizeCSS](https://mantine.dev/theming/global-styles/) components to include global styles
- New [Styles API](https://mantine.dev/theming/styles-api/) allows you to style any part of Mantine components with inline styles or classes

## New @mantine/dates package

@mantine/dates s a new package that includes date pickers and calendars, it includes components:
- [DatePicker](https://mantine.dev/dates/date-picker/)
- [DateRangePicker](https://mantine.dev/dates/date-range-picker/)
- [TimeInput](https://mantine.dev/dates/time-input/)
- [Calendar](https://mantine.dev/dates/calendar/)
- [RangeCalendar](https://mantine.dev/dates/range-calendar/)
- [Month](https://mantine.dev/dates/month/)

## New @mantine/prism package

New [@mantine/prism](https://mantine.dev/others/prism/) package exports `Prism` component that allows you to highlight code with [prism-react-renderer](https://github.com/FormidableLabs/prism-react-renderer) and your theme colors just like in Mantine docs.

## New @mantine/hooks hooks

- [use-did-update](https://mantine.dev/hooks/use-did-update) – useEffect replacement when you do not want hook to be called on mount
- [use-isomorphic-effect](https://mantine.dev/hooks/use-isomorphic-effect) – switch between useEffect during server side rendering and useLayoutEffect after hydration
- [use-force-update](https://mantine.dev/hooks/use-force-update) – force component to rerender
- [use-toggle](https://mantine.dev/hooks/use-toggle) – switch state between two values
- [use-uncontrolled](https://mantine.dev/hooks/use-uncontrolled) – manage state for both controlled and uncontrolled components
- [use-window-scroll](https://mantine.dev/hooks/use-window-scroll) – get current scroll position and handler to scroll to given position

## New @mantine/core components

- [Select](https://mantine.dev/core/select/) – custom searchable select with option to use custom option component
- [Affix](https://mantine.dev/core/affix/)
- [Accordion](https://mantine.dev/core/accordion/)
- [Autocomplete](https://mantine.dev/core/autocomplete/)
- [JsonInput](https://mantine.dev/core/json-input/) 
- [Slider and RangeSlider](https://mantine.dev/core/slider/)

## @mantine/core changes

- [Tabs](https://mantine.dev/core/tabs/) component now supports unstyled and pills variants
- All inputs now support `size` prop
- [Drawer](https://mantine.dev/core/drawer/) component now support title and close button
- [Tooltip](https://mantine.dev/core/tooltip/) component now displays tooltip on focus by default
- [Menu](https://mantine.dev/core/menu/) component now closes on scroll to prevent incorrect focus handling
- [Menu](https://mantine.dev/core/menu/)  component now supports hover trigger
- 3 new premade transitions were added to [Transition component](https://mantine.dev/core/menu/): scale-y, scale-x and scale
- @mantine/core now exports new [UnstyledButton](https://mantine.dev/core/button/#unstyled-button)  component that can be used to create custom buttons

<hr /><em>This discussion was created from the release <a href='https://github.com/mantinedev/mantine/releases/tag/2.0.0'>2.0.0</a>.</em>

**Comment by cstrat** [comment]
First of all amazing update!
Thanks for all the work thats gone into it.

I've played with `Autocomplete` and the new `Select` - brilliant!
Will provide feedback as I test.
